code/src/nuvla/connector/kubernetes.py

```diff
@@ -169,15 +169,84 @@ def list(self, filters=None):
         # return execute_cmd(cmd)
         pass
 
+    def _get_container_logs(self, namespace, values, since_opt, lines: int) -> str:
+        tail_lines=lines # wtf
+        tail_lines=5 # wtf
+        logs_string=str()
+        pod_unique_id = str(values["metadata"]["name"])
+        log.debug('Unique pod ID: %s ', pod_unique_id)
+        try:
+            for containers_list in values['spec']['containers']:
+                container = str(containers_list['name'])
+                log.debug('Found container: %s ', container)
+                list_opts_log = ['--timestamps=true', '--tail', str(tail_lines),
+                                 '--namespace', namespace] + since_opt
+                container_opts = \
+                ['pod/' + pod_unique_id, '--container=' + container]
+                cmd = \
+                self.build_cmd_line(['logs'] + container_opts + list_opts_log)
+                log.debug('Generated logs command line : %s', cmd)
+                header_line = "\n\nLog last " + str(tail_lines) + \
+                " lines for Container " + \
+                container + " in Pod " + pod_unique_id + " \n"
+                log.debug('Header line : %s', header_line)
+                logs_string = \
+                logs_string + header_line + execute_cmd(cmd).stdout
+                log.debug('_get_container_logs logs string : %s', logs_string)
+        except Exception as e_cont:
+            ex_string = "There was a problem getting logs from Pod " \
+            + pod_unique_id
+            log.debug('An exception caught: %s ', e_cont)
+            logs_string = str(ex_string)
+        return logs_string 
+
+    def _get_the_pods(self, namespace, values, since_opt, lines: int) -> str:
```

Comment:
Not clear what is `values`. Is it a list of all discovered pods? Then IMO `pods: dict` would be a better name.
The method is called `_get_the_pods` however, what it does is get logs from all the containers of all the pods. It needs to be renamed. Then, update the logging string accordingly. Now it refers to `_get_containers_logs`.